core/arch/arch.c

```diff
@@ -3355,7 +3355,8 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
      * if we append to dr_mcontext_t in the future we'll need
      * to check src->size here.
      */
-    if (src->size != sizeof(dr_mcontext_t))
+    if (src->size != sizeof(dr_mcontext_t) &&
+        src->size != sizeof(dr_mcontext_t) - sizeof(dr_opmask_t))
```

Comment:
I would expect this check to just completely go away: there's no reason to enumerate sizes twice, and the list of sizes is going to grow.  I would remove this and check all new sizes below as part of copying fields.

A good example of handling struct size fields is dr_get_os_version().  Unfortunately widening existing fields, instead of just adding new ones, makes this more complex.  Adding new fields instead of widening would presumably have an overhead cost on copying to and from.

Reply:
Done, except I kept the old check like this: if (size > sizeof(dr_mcontext_t)) return false;